### PR TITLE
Support applications without JMenu.

### PR DESCRIPTION
**Pull request by realityking**
Deprecate some code not needed with the autoloader.

### CODE REVIEW

File: libraries/legacy/menu/menu.php [hunk 1]
```diff
@@ -112,7 +114,8 @@ public static function getInstance($client, $options = array())
 			}
 			else
 			{
-				throw new Exception(JText::sprintf('JLIB_APPLICATION_ERROR_MENU_LOAD', $client), 500);
```
Comment (chdemko) [inline libraries/legacy/menu/menu.php]: why are you removing this line since you don't remove it router.php?

Comment (realityking) [inline libraries/legacy/menu/menu.php]: That's intentional. If you look in the old code for the 3 classes, JPathway and JRouter used to return a JError object but JMenu returned null. Without this the CMS installation actually fails to work.

We could deal with this in JApplication of course, I thought this was desirable for B/C but I wouldn't mind changing it.

**Comment by realityking** [comment]
Added some fixes for things I overlooked the first time. We really should catch these exceptions in JApplication. Now we'd be able to deal with JMenu::getInstance() throwing exceptions as well, but I'm not sure of the B/C impact.

**Comment by piotr-cz** [comment]
Yes!

**Pull request merged**